java/src/main/java/com/saucelabs/simplesauce/SauceSession.java

```
@@ -35,43 +35,45 @@ public MutableCapabilities getSauceOptionsCapability(){
     public SauceSession() {
         currentSessionCapabilities = new MutableCapabilities();
         sauceDriver = new SauceDriverImpl();
-        environmentManager = new EnvironmentManagerImpl();
         sauceOptions = new SauceOptions();
     }
 
-    public SauceSession(SauceRemoteDriver remoteManager, EnvironmentManager environmentManager) {
+    public SauceSession(SauceRemoteDriver remoteManager) {
         sauceDriver = remoteManager;
         currentSessionCapabilities = new MutableCapabilities();
-        this.environmentManager = environmentManager;
         sauceOptions = new SauceOptions();
     }
 
     public SauceSession(SauceOptions options) {
         sauceOptions = options;
         currentSessionCapabilities = new MutableCapabilities();
-        environmentManager = new EnvironmentManagerImpl();
         sauceDriver = new SauceDriverImpl();
     }
 
-    public SauceSession(SauceOptions options, SauceRemoteDriver remoteManager, EnvironmentManager environmentManager) {
+    public SauceSession(SauceOptions options, SauceRemoteDriver remoteManager) {
         sauceOptions = options;
         sauceDriver = remoteManager;
         currentSessionCapabilities = new MutableCapabilities();
-        this.environmentManager = environmentManager;
     }
 
     public WebDriver start() {
-        mutableCapabilities = appendSauceCapabilities();
-        setBrowserSpecificCapabilities(sauceOptions.getBrowserName());
-        currentSessionCapabilities = setRemoteDriverCapabilities(mutableCapabilities);
-        tryToCreateRemoteWebDriver(sauceDataCenter);
-        return webDriver;
+        if (getSauceUsername() == null) {
```

This `start()` method got hairy really quick with the if else clauses. Any reason you don't want to just keep this login in the corresponding methods where they existed here?
```
        mutableCapabilities.setCapability("username", getUserName());	        mutableCapabilities.setCapability("username", getSauceUsername());
        mutableCapabilities.setCapability("accessKey", getAccessKey());
```

---

yeah, for security reasons it needs to go into the URL
Current implementation effectively does the evaluation in the start method, but puts it in the getter, and I don't think a getter should throw an exception. If there isn't a username set it should return null

Now, what we probably should do is to have the public setter/getter be `username` / `accessKey` and have `getSauceUsername` and `getSauceAccessKey` be private methods that throw this error. Does that sound good?

---

ok, I implemented what I'm thinking in the latest commit